server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementIterator.java

```diff
@@ -260,11 +260,6 @@ private void computeTabletManagementActions(final TabletMetadata tm,
       reasonsToReturnThisTablet.add(ManagementAction.NEEDS_VOLUME_REPLACEMENT);
     }
 
-    if (!reasonsToReturnThisTablet.isEmpty()) {
```

Comment:
I think I suggested this change in some other PR.  Its causing multiple problems. I tried moving it later in the code and  that fixed the volume recovery problem, but it still caused problems with root tablet compaction.